test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ActionOverloadingTests/Tests/ActionOverloadingQueryTests.cs

```diff
@@ -0,0 +1,253 @@
+                services.ConfigureControllers(typeof(ProductsController), typeof(OrderLinesController), typeof(PeopleController), typeof(MetadataController));
+
+                services.AddControllers().AddOData(opt => opt.Count().Filter().Expand().Select().OrderBy().SetMaxTop(null)
+                    .AddRouteComponents("odata", CommonEndToEndEdmModel.GetEdmModel()));
+            }
+        }
+
+        public ActionOverloadingQueryTests(TestWebApplicationFactory<TestsStartup> fixture)
+            : base(fixture)
+        {
+            _baseUri = new Uri(Client.BaseAddress, "odata/");
+            _context = new Container(_baseUri);
+            _context.HttpClientFactory = HttpClientFactory;
+            _model = CommonEndToEndEdmModel.GetEdmModel();
+        }
+
+        /// <summary>
+        /// Verify metadata of the operations defined in the test service.
+        /// </summary>
+        [Fact]
+        public void QueryMetadataTest()
+        {
+            Dictionary<string, string> expectedUpdatePersonInfoOperations = new Dictionary<string, string>()
+            {
+                { "", null }, // the unbound action
+            };
+            Dictionary<string, string> expectedRetrieveProductOperations = new Dictionary<string, string>()
+            {
+                { "", null }, // the service operation
+            };
+
+            IEnumerable<IEdmOperationImport> actionImports = _model.EntityContainer.OperationImports();
+
+            this.VerifyOperationsInMetadata(expectedUpdatePersonInfoOperations, actionImports.Where(fi => fi.Name == "UpdatePersonInfo"));
+            this.VerifyOperationsInMetadata(expectedRetrieveProductOperations, actionImports.Where(fi => fi.Name == "RetrieveProduct"));
+        }
+
+        /// <summary>
+        /// Verify actions in entry payload format atom, json verbose, and json fullmetadata.
+        /// </summary>
+        [Fact]
+        public void QueryEntryTest()
+        {
+            List<string> mimeTypes = new List<string>()
+            {
+                //MimeTypes.ApplicationAtomXml,
+                MimeTypes.ApplicationJson + MimeTypes.ODataParameterFullMetadata,
+            };
+
+            string updatePersonInfo = "UpdatePersonInfo";
+            string updatePersonInfoName = "Default.UpdatePersonInfo";
+            string retrieveProduct = "RetrieveProduct";
+            string retrieveProductName = "Default.RetrieveProduct";
+            string increaseEmployeeSalary = "IncreaseEmployeeSalary";
+            string increaseEmployeeSalaryName = "Default.IncreaseEmployeeSalary";
+            string sack = "Sack";
+            string sackName = "Default.Sack";
+
+            foreach (string mimeType in mimeTypes)
+            {
+                List<Tuple<string, string>> expectedActionsOnPerson = new List<Tuple<string, string>>()
+                {
+                    new Tuple<string, string>(MetadataPrefix + updatePersonInfo, "People(-1)/" + updatePersonInfoName),

```

Comment:
Consider the shorthand ValueTuple syntax:
```suggestion
                    (MetadataPrefix + updatePersonInfo, "People(-1)/" + updatePersonInfoName),

```